.github/workflows/test.yml

```
@@ -0,0 +1,48 @@
+name: Test
+
+on:
+  push:
+    branches: [ main, master ]
+  pull_request:
+    branches: [ main, master ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    
+    strategy:
+      matrix:
+        go-version: [1.20, 1.21, 1.22]
+    
+    steps:
+    - uses: actions/checkout@v4
```

there's a newer version of this action
```suggestion
    - uses: actions/checkout@v5
```